pages/lessons/projects/6.mdx

```diff
@@ -0,0 +1,796 @@
+  Solidity code, as well as a set of built-in commands for working with and
+  debugging your code, which makes it a useful tool for quickly testing and
+  experimenting with Solidity code without having to spin up a full foundry
+  project
+
+Now we can create our project with:
+
+```bash
+forge init my_token
+```
+
+Let's see what our project looks like:
+
+```bash
+cd my_token
+tree . -d -L 1
+```
+
+This command outputs something like this:
+
+```bash
+.
+├── lib
+├── script
+├── src
+└── test
+
+4 directories
+```
+
+The folders in our project are used like this:
+
+- `lib` _dependencies_: Forge uses git submodules instead of Node
+  Packages. This means it works with any GitHub repository that contains smart
+  contracts.
+- `script` scripts: automate transactions or deployment or our smart contracts
+  (scripts are written in solidity!)
+- `src` source: where our smart contract code will live
+- `test` tests: write tests for our smart contracts (also in solidity!)
+
+The default project includes a basic smart contract. We can build it to
+see how forge commands work:
+
+```bash
+forge build
+```
+
+My output:
+
+```bash
+[⠊] Compiling...
+[⠑] Installing solc version 0.8.19
+[⠊] Successfully installed solc 0.8.19
+[⠒] Compiling 20 files with 0.8.19
+[⠑] Solc 0.8.19 finished in 6.08s
+Compiler run successful
+```
+
+And now we can test it:
+
+```bash
+forge test
+```
+
+My output:
+
+```bash
+[⠰] Compiling...
+No files changed, compilation skipped
+
+Running 2 tests for test/Counter.t.sol:CounterTest
+[PASS] testIncrement() (gas: 28334)
+[PASS] testSetNumber(uint256) (runs: 256, μ: 27553, ~: 28409)
+Test result: ok. 2 passed; 0 failed; finished in 19.11ms
+```
+
+As we mentioned earlier, we are going to use OpenZeppelin ERC20 implementation.
+To use it, we need to install OpenZeppelin as a dependency in our project. To
+install dependencies in Foundry, we can use:
+
+```bash
+forge install OpenZeppelin/openzeppelin-contracts
+```
+
+This looks for the repo called `openzeppelin-contracts` from the GitHub user
+`OpenZeppelin` and downloads it as a git submodule.
+
+This is the output I got:
+
+```bash
+Installing openzeppelin-contracts in "/home/myuser/d_d_academy/my_token/lib/openzeppelin-contracts" (url: Some("https://github.com/OpenZeppelin/openzeppelin-contracts"), tag: None)
+    Installed openzeppelin-contracts v4.8.0
+```
+
+We can now import OpenZeppelin contracts in our smart contracts, but we can go
+the extra-mile to use the same import line we used in Hardhat before.
+We need to create a `remappings.txt` file in the root of our project:
+
+```bash
+forge remappings > remappings.txt
+echo "@openzeppelin/=lib/openzeppelin-contracts/" >> remappings.txt
+```
+
+This tells Forge, “Hey, anytime you hit `@openzepplin/`, look in
+`lib/openzeppelin-contracts/` instead.”
+
+Now that we have an overview of the tools and the OpenZeppelin's libraries
+installed, we can clean up the files we won't use and start fresh:
+
```

Comment:
Maybe clarify that the 3 commands below can be done in one shot or specify that they might be done separately?

Reply:
I prefer to keep instructions clean, and if there's someone with more advanced terminal knowledge, they can save as much time as they want.
I was trying to focus more on the Solidity content than on teaching how to use the terminal.